pkg/datamover/utils.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datamover
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"os"
+
+	"github.com/pkg/errors"
+
+	"github.com/kanisterio/kanister/pkg/kopia/snapshot"
+	"github.com/kanisterio/kanister/pkg/location"
+	"github.com/kanisterio/kanister/pkg/output"
+	"github.com/kanisterio/kanister/pkg/param"
+)
+
+const (
+	usePipeParam = `-`
+)
+
+func targetWriter(target string) (io.Writer, error) {
+	if target != usePipeParam {
+		return os.OpenFile(target, os.O_RDWR|os.O_CREATE, 0755)
+	}
+	return os.Stdout, nil
+}
+
+func locationPull(ctx context.Context, p *param.Profile, path string, target io.Writer) error {
+	return location.Read(ctx, target, *p, path)
+}
+
+// kopiaLocationPull pulls the data from a kopia snapshot into the given target
+func kopiaLocationPull(ctx context.Context, backupID, path, targetPath, password string) error {
+	switch targetPath {
+	case usePipeParam:
+		return snapshot.Read(ctx, os.Stdout, backupID, path, password)
+	default:
+		return snapshot.ReadFile(ctx, backupID, targetPath, password)
+	}
+}
+
+func secretsFromRepositoryServerCR(rs *param.RepositoryServer) (string, string, string, error) {
```

Comment:
Can we find a better package for this helper? It seems like it's just getting all the secrets from a repo server CR which is not related to the `DataMover` interface

Reply:
Done !